### PR TITLE
Implemtent the multi-line string method

### LINKED ISSUE #167: Add multi-line string
The specification for multi-line strings is the same as in Python. In multi-line strings, quote is done with """ and line breaks are allowed.

It would be relatively easy to implement by combining multi-line comments and string parsing in lex.rs.

### PR DESCRIPTION
**Pull request by GreasySlug**
Fixes #167.

Add method for multi-line strings and a file to test it.

A test for tokenizing multi-line strings will also be added.

Changes proposed in this PR:
- Add lex_multi_line_str() method
- Add multi-line string and string conditional branching
- Add test to ensure that the added methods function properly

@mtshiba

### CODE REVIEW

File: compiler/erg_parser/lex.rs [hunk 1]
```diff
@@ -688,6 +688,96 @@ impl Lexer /*<'a>*/ {
...
+                                if self.peek_next_ch().is_some()
+                                    && self.peek_next_ch().unwrap() == '\n'
+                                {
+                                    continue; // Escaping a line break if only '\' comes at the end
+                                }
+                            }
+                            '"' => s.push('"'),
+                            't' => s.push_str("    "), // tab is invalid, so changed into 4 whitespace
+                            '\\' => s.push('\\'),
+                            'n' => s.push('\n'),
```
Comment (GreasySlug) [inline compiler/erg_parser/lex.rs:738]: Do I need to add "self.lineno..." below here as well?

Comment (mtshiba) [reply compiler/erg_parser/lex.rs:738]: Yes, I am. This is required for the location of the token.

Comment (GreasySlug) [reply compiler/erg_parser/lex.rs:738]: I will add this and a test for tokenization.

**Comment by GreasySlug** [comment]
I found a new bug in implementing this.
Tokenized for quotes only, but trimmed after execution.
```sh
>>> a = """\"\"\""""

>>> print! a

None
```

**Comment by mtshiba** [comment]
Thank you! We will deal with the bug in another issue.

**Pull request merged**